pkg/fleet-manager/backup_controller.go

```diff
@@ -0,0 +1,628 @@
+		case "restore":
+			return b.reconcileDeleteRestore(ctx, currentObject.(*backupapi.Restore))
+		case "migrate":
+			return b.reconcileDeleteMigrate(ctx, currentObject.(*backupapi.Migrate))
+		}
+	}
+
+	// Handle the main reconcile logic based on the type of the CRD object found
+	switch crdType {
+	case "backup":
+		return b.reconcileBackup(ctx, currentObject.(*backupapi.Backup))
+	case "restore":
+		return b.reconcileRestore(ctx, currentObject.(*backupapi.Restore))
+	case "migrate":
+		return b.reconcileMigrate(ctx, currentObject.(*backupapi.Migrate))
+	}
+
+	log.Error(errors.New("unreachable code reached"), "This should not happen")
+	return ctrl.Result{}, nil
+}
+
+// reconcileBackup handles the main reconcile logic for a Backup object.
+func (b *BackupManager) reconcileBackup(ctx context.Context, backup *backupapi.Backup) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch destination clusters
+	destinationClusters, err := b.fetchDestinationClusters(ctx, backup.Namespace, backup.Spec.Destination)
+	if err != nil {
+		log.Error(err, "failed to fetch destination clusters for backup", "backupName", backup.Name)
+		return ctrl.Result{}, err
+	}
+	// Apply velero backup resource in target clusters
+	result, err := b.reconcileBackupResources(ctx, backup, destinationClusters)
+	if err != nil || result.Requeue || result.RequeueAfter > 0 {
+		return result, err
+	}
+	// Collect velero backup resource status to current backup
+	return b.reconcileBackupStatus(ctx, backup, destinationClusters)
+}
+
+// reconcileBackupResources converts the backup resources into velero backup resources that can be used by Velero on the target clusters, and applies each of these backup resources to the respective target clusters.
+func (b *BackupManager) reconcileBackupResources(ctx context.Context, backup *backupapi.Backup, destinationClusters map[ClusterKey]*fleetCluster) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	backupLabel := generateVeleroInstanceLabel(BackupNameLabel, backup.Name, backup.Spec.Destination.Fleet)
+
+	if isScheduleBackup(backup) {
+		// Handle scheduled backups
+		for clusterKey, fleetCluster := range destinationClusters {
+			veleroScheduleName := generateVeleroResourceName(clusterKey.Name, BackupKind, backup.Name)
+			veleroSchedule := buildVeleroScheduleInstance(&backup.Spec, backupLabel, veleroScheduleName)
+			if err := createVeleroScheduleInstance(fleetCluster, veleroSchedule); err != nil {
+				log.Error(err, "failed to create velero schedule instance for backup", "backupName", backup.Name)
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		// Handle one time backups
+		for clusterKey, fleetCluster := range destinationClusters {
+			veleroBackupName := generateVeleroResourceName(clusterKey.Name, BackupKind, backup.Name)
+			veleroBackup := buildVeleroBackupInstance(&backup.Spec, backupLabel, veleroBackupName)
+			if err := createVeleroBackupInstance(fleetCluster, veleroBackup); err != nil {
```

Comment:
we can abstract and reuse `ApplicationManager.syncResource`

Reply:
ok, I will try it